src/diffpy/utils/diffraction_objects.py

```diff
@@ -390,14 +390,15 @@ def scale_to(self, target_diff_object, xtype=None, xvalue=None):
 
         data = self.on_xtype(xtype)
         target = target_diff_object.on_xtype(xtype)
+        if len(data[0]) == 0 or len(target[0]) == 0 or len(data[0]) != len(target[0]):
+            raise ValueError("I cannot scale two diffraction objects with empty or different lengths.")
```

Comment:
@sbillinge I added this error message here but I'm not sure if this is appropriate. Does it make more sense that we find `xindex` for each diffraction object?